### PR TITLE
Update TX pages for API parity

### PR DESCRIPTION
**Pull request by shorsher**
- Remove deprecated fields from UI panels
 - Updated TX details to become a `<Slide>` and include transaction status objects
 - Remove links to TX details in Messages view
Screen Shot 2022-01-31 at 4 10 48 PM
Screen Shot 2022-01-31 at 4 15 59 PM
Screen Shot 2022-01-31 at 4 15 50 PM

### CODE REVIEW

File: src/core/interfaces.ts [hunk 1]
```diff
@@ -44,6 +44,13 @@ export enum FFColors {
   Red = '#FF0000',
 }
 
+export enum TransactionType {
+  None = 'none',
+  BatchPin = 'batch_pin',
+  TokenPool = 'token_pool',
+  TokenTransfer = 'token_transfer',
+}
```
Comment (awrichar) [inline src/core/interfaces.ts:52]: There is a new 'contract_invoke' in mainline at this point too

File: src/modules/data/views/Transactions/TransactionList.tsx [hunk 2]
```diff
@@ -52,13 +50,7 @@ export const TransactionList: React.FC<Props> = ({ filterString }) => {
...
   const { createdFilter, lastEvent } = useContext(ApplicationContext);
 
-  const columnHeaders = [
-    t('hash'),
-    t('blockNumber'),
-    t('signer'),
-    t('status'),
-    t('dateMined'),
-  ];
+  const columnHeaders = [t('id'), t('type'), t('status'), t('dateMined')];
```
Comment (awrichar) [inline src/modules/data/views/Transactions/TransactionList.tsx:53]: "dateMined" is distinctly incorrect at this point ("Created" is probably more appropriate)

File: src/modules/data/views/Transactions/TransactionList.tsx [hunk 3]
```diff
@@ -132,27 +124,15 @@ export const TransactionList: React.FC<Props> = ({ filterString }) => {
...
-        },
-        { value: tx.info?.blockNumber },
-        {
-          value: (
-            <HashPopover textColor="secondary" address={tx.subject.signer} />
-          ),
-        },
+        { value: <HashPopover address={tx.id} shortHash /> },
+        { value: tx.type },
         { value: tx.status },
```
Comment (awrichar) [inline src/modules/data/views/Transactions/TransactionList.tsx:129]: Status is also about to be removed

**Review by awrichar — APPROVED** [review]
Approved, but requested a few small tweaks for this or a follow-up

File: src/modules/data/views/Transactions/TransactionList.tsx (hunk 3 above)
Comment (awrichar) [reply src/modules/data/views/Transactions/TransactionList.tsx:129]: See https://github.com/hyperledger/firefly/pull/457

**Comment by shorsher** [comment]
I wasn't sure if `Status` was the right word to use in the panel, let me know if you have any better suggestions.

File: src/modules/data/views/Transactions/TransactionDetails.tsx [hunk 4]
```diff
@@ -14,243 +14,153 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
```
Comment (eberger727) [inline src/modules/data/views/Transactions/TransactionDetails.tsx:16]: image
Remove block is id is undefined

**Review by eberger727 — APPROVED** [review]
Looks good, requested one change

File: src/core/interfaces.ts [hunk 5]
```diff
@@ -150,18 +158,22 @@ export interface INamespace {
...
+  type: TransactionType;
+}
+
+export interface ITransactionStatus {
+  status: string;
+  details: {
+    error: string;
+    id: string;
+    status: string;
+    subtype: string;
```
Comment (awrichar) [inline src/core/interfaces.ts:172]: error, id, and subtype are all optional (may be undefined)

File: src/core/interfaces.ts [hunk 6]
```diff
@@ -150,18 +158,22 @@ export interface INamespace {
...
-  info?: IEthTransactionInfo;
-  subject: {
-    signer: string;
-    namespace: string;
+  namespace: string;
+  type: TransactionType;
+}
+
+export interface ITransactionStatus {
+  status: string;
```
Comment (awrichar) [inline src/core/interfaces.ts:167]: status (here and in the details) is an enum of Pending/Failed/Succeeded, so might be useful to define that

**Comment by awrichar** [comment]
@shorsher  I think "Status" is correct as the label for Pending/Succeeded/Failed, but I do think it's confusing as the header for the child objects. Perhaps just "Details" for that header?

File: src/core/components/TransactionStatus.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  status: ITransactionStatus;
+}
+
+export const TransactionStatus: React.FC<Props> = ({ status }) => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+
+  return (
+    <>
+      <Typography fontWeight="bold">{t('status')}</Typography>
```
Comment (awrichar) [inline src/core/components/TransactionStatus.tsx:35]: Commented on the top-level too... but maybe "Details" is a more appropriate header here

**Pull request merged**